source/world_builder/features/plume.cc

```diff
@@ -0,0 +1,346 @@
+
+                          }
+                        break;
+                        case 2: // composition
+
+                          for (const auto &composition_model: composition_models)
+                            {
+                              output[entry_in_output[i_property]] = composition_model->get_composition(position_in_cartesian_coordinates,
+                                                                                                       position_in_natural_coordinates,
+                                                                                                       depth,
+                                                                                                       properties[i_property][1],
+                                                                                                       output[entry_in_output[i_property]],
+                                                                                                       min_depth_local,
+                                                                                                       max_depth_local);
+
+                              WBAssert(!std::isnan(output[entry_in_output[i_property]]), "Composition is not a number: " << output[entry_in_output[i_property]]
+                                       << ", based on a composition model with the name " << composition_model->get_name() << ", in feature " << this->name);
+                              WBAssert(std::isfinite(output[entry_in_output[i_property]]), "Composition is not finite: " << output[entry_in_output[i_property]]
+                                       << ", based on a composition model with the name " << composition_model->get_name() << ", in feature " << this->name);
+
+                            }
+
+                          break;
+                        }
+                      case 3: // grains
+                      {
+                        WorldBuilder::grains  grains(output,properties[i_property][2],entry_in_output[i_property]);
+                        for (const auto &grains_model: grains_models)
+                          {
+                            grains = grains_model->get_grains(position_in_cartesian_coordinates,
+                                                              position_in_natural_coordinates,
+                                                              depth,
+                                                              properties[i_property][1],
+                                                              grains,
+                                                              min_depth_local,
+                                                              max_depth_local);
+
+                          }
+                        grains.unroll_into(output,entry_in_output[i_property]);
+                      }
+                      break;
```

Comment:
It now also needs a case 4 and a tag. See 
https://github.com/GeodynamicWorldBuilder/WorldBuilder/blob/dfbb30bbb8164ea46f52a8f787ed964bcc8cde7d/source/world_builder/features/continental_plate.cc#L234-L244

and 

https://github.com/GeodynamicWorldBuilder/WorldBuilder/blob/dfbb30bbb8164ea46f52a8f787ed964bcc8cde7d/source/world_builder/features/continental_plate.cc#L89-L94